Bluejay/Bluejay/Data+Extractable.swift

```diff
@@ -17,7 +17,7 @@ extension Data {
         - start: The starting position of the range to read.
         - length: The number of bytes to read from `start`.
     */
-    public func extract<T>(start: Int, length: Int) -> T {
+    public func extract<T>(start: Int, length: Int) throws -> T {
         return self.subdata(in: start..<start+length).withUnsafeBytes { $0.pointee }
     }
```

Comment:
I think we actually want to have a parse error that gets thrown here, rather than just making it throwable, right? It's not like the throwable init where we are providing that mainly so that clients can make Receivable implementing classes that throw, here we want it to actually throw if you can't extract the required data. 

Something like 
```
public func extract<T>(start: Int, length: Int) throws -> T {
    if start + length > data.length {
        throw ParseError.notEnoughData
    }

    return self.subdata(in: start..<start+length).withUnsafeBytes { $0.pointee }		          return self.subdata(in: start..<start+length).withUnsafeBytes { $0.pointee }
}
```

Reply:
Yes that's right. I've added the fix and the new error.